src/main/java/com/tabnine/binary/fetch/BootstrapperSupport.java

```diff
@@ -0,0 +1,73 @@
+package com.tabnine.binary.fetch;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.util.text.SemVer;
+import com.tabnine.general.StaticConfig;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.prefs.Preferences;
+
+import static java.lang.String.format;
+
+public class BootstrapperSupport {
+    static Optional<BinaryVersion> bootstrap_version(LocalBinaryVersions localBinaryVersions, BinaryRemoteSource binaryRemoteSource, BinaryDownloader binaryDownloader) {
```

Comment:
Java naming convention for method names is without underscores, like so: `bootstrapVersion`. But I would like to suggest you use a better name for it that include some verb like `findVersion`

Reply:
rust habit :)